src/libs/ajax/metrics/useMetrics.ts

```diff
@@ -0,0 +1,13 @@
+import { useMemo } from 'react'
+import { Ajax } from 'src/libs/ajax'
+
+
+export type CaptureEventFn = (event: string, details?: {}) => void
+
+export const useMetricsEvent = (): CaptureEventFn => {
```

Comment:
I don't understand the point of this hook.

> implemented useMetrics helper hook to remove boiler-plate code from event logging and improve Typescript ergonomics.

This seems like it adds boilerplate. Now I have to call the hook to get a function and then call that function instead of directly calling a function.

```js
const captureEvent = useMetricsEvent()
captureEvent(...)
```

```js
Ajax().Metrics.captureEvent(...)
```

Reply:
I copied that bullet point from an earlier commit in my local work.  Some of the boilerplate was transitory (Typescript cruft that the final implementation removes).  But it:
- removes repetitive Ajax() constructor-like calls (efficiency win)
- is shorter in the main code by removing Ajax().Metrics prefix
- far easier to mock, and removes unit-test boilerplate required to mock anything from Ajax() cluster.
- moves us to a more explicit dependency pattern instead of just Ajax().everythingUnderTheSun()

Reply:
Could we accomplish all that and avoid the need for a hook by exporting `captureEvent` from Metrics.js?

Reply:
Looks like I edited the comment above just after you read it... 
But to your follow-up question... captureEvent is currently nested inside Ajax().Metrics sub-object's contract... I think your suggestion would break that pattern... which I don't think we want to just do willy-nilly.  But also, that would not bring over these benefits of the hook:
- wrapping void await to avoid ts/lint adherence cruft by consumers
- cleaner dependency and mocking pattern

Reply:
The wrapper could be exported. And that would still improve mocking since you wouldn't have to mock the entire Ajax contract.

Reply:
Ajax(signal).subContract pattern allows for Ajax(signal) instantiation-like flow-thru and other functional composition that I don't think we want to circumvent at this point.  While consumers of this specific function do not appear to make use of signal for cancelability,  a goal of this solution is to be conservative and additive for now.  Later discussion and proposals around code organization and patterns will come this year.

Also, (at the risk of sounding pedantic) choosing to fire-and-forget the promise feels like an application decision... These Ajax sub-contracts are pure in that they just expose promise-based contracts, leaving other concerns for elsewhere to determine.

I may be misinterpreting your suggestion though.

Reply:
as we discussed on the call, another goal generally is to align dependency usage in our components into the pattern of useIntrestingData, useLogger, etc...  having a cleaner and narrower decoupling of each dependency... which is an iterative step towards key underpinnings of the larger cross-team code sharing initiative we are tackling this year.

Reply:
Discussed in person. I was missing the bit that `useMetrics` separates components from the code in `libs/ajax` and `useMetrics` could be updated in the future to allow replacing the implementation of `captureEvent` via context, etc.